improver_tests/psychrometric_calculations/condensation_trails/test_ContrailHeightExtractor.py

```diff
@@ -0,0 +1,309 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Unit tests for the ContrailHeightExtractor plugin"""
+
+import iris
+import numpy as np
+import pytest
+
+from improver.psychrometric_calculations.condensation_trails import (
+    ContrailHeightExtractor,
+)
+
+
+def make_cube(
+    data,
+    name,
+    units,
+    dims=("pressure", "latitude", "longitude"),
+    contrail_type_values=None,
+    contrail_type_meaning_values=None,
+):
+    """
+    Simple cube factory for ContrailsHeightExtractor tests.
+
+    Args:
+        data (np.ndarray): Data array.
+        name (str): Name of the cube.
+        units (str): Units of the cube.
+        contrail_type_values (list): Values for the contrail_type attribute, e.g. [0, 1, 2]
+        contrail_type_meaning_values (str): Meanings for the contrail_type attribute, e.g. "None Non-persistent Persistent"
+        dims (tuple): Dimensions of the cube.
+
+    Returns:
+        iris.cube.Cube: The constructed cube.
+    """
+    shape = data.shape
+    coords = []
+    if "engine_factor" in dims:
+        coords.append(("engine_factor", np.arange(shape[0])))
+        if "pressure" in dims:
+            coords.append(("pressure", np.arange(shape[1])))
+        if "latitude" in dims:
+            coords.append(("latitude", np.arange(shape[2])))
+        if "longitude" in dims:
+            coords.append(("longitude", np.arange(shape[3])))
+    else:
+        if "pressure" in dims:
+            coords.append(("pressure", np.arange(shape[0])))
+        if "latitude" in dims:
+            coords.append(("latitude", np.arange(shape[1])))
+        if "longitude" in dims:
+            coords.append(("longitude", np.arange(shape[2])))
+    cube = iris.cube.Cube(data, long_name=name, units=units)
+    for dim, (coord_name, points) in enumerate(coords):
+        if coord_name == "engine_factor":
+            cube.add_dim_coord(
+                iris.coords.DimCoord(points, long_name="engine_factor"), dim
+            )
```

Comment:
```suggestion
    if "engine_contrail_factor" in dims:
        coords.append(("engine_contrail_factor", np.arange(shape[0])))
        if "pressure" in dims:
            coords.append(("pressure", np.arange(shape[1])))
        if "latitude" in dims:
            coords.append(("latitude", np.arange(shape[2])))
        if "longitude" in dims:
            coords.append(("longitude", np.arange(shape[3])))
    else:
        if "pressure" in dims:
            coords.append(("pressure", np.arange(shape[0])))
        if "latitude" in dims:
            coords.append(("latitude", np.arange(shape[1])))
        if "longitude" in dims:
            coords.append(("longitude", np.arange(shape[2])))
    cube = iris.cube.Cube(data, long_name=name, units=units)
    for dim, (coord_name, points) in enumerate(coords):
        if coord_name == "engine_contrail_factor":
            cube.add_dim_coord(
                iris.coords.DimCoord(points, long_name="engine_contrail_factor"), dim
            )
```
The output cube from `CondensationTrailFormation` will have `engine_contrail_factor` as its first coordinate, so this should be reflected in the tests to avoid confusion.